value_modifier_test.go

```diff
@@ -0,0 +1,346 @@
+		Input          string
+		Expected       string
+	}{
+		// Test case 1, a single modifier modifies all secrets.
+		{
+			ValueModifiers: []ValueModifier{
+				testModifier1{},
+			},
+			IgnoreFields: []string{},
+			Input: `{
+  "noSecret1": "noSecret1",
+  "pass1": "pass1"
+}`,
+			Expected: `{
+  "noSecret1": "noSecret1-modified1",
+  "pass1": "pass1-modified1"
+}`,
+		},
+		// Test case 2, a single modifier modifies all numeric secrets.
+		{
+			ValueModifiers: []ValueModifier{
+				testModifier1{},
+			},
+			IgnoreFields: []string{},
+			Input: `{
+  "noSecret1": "noSecret1",
+  "pass1": 12345
+}`,
+			Expected: `{
+  "noSecret1": "noSecret1-modified1",
+  "pass1": "12345-modified1"
+}`,
+		},
+		// Test case 3, a single modifier modifies all secrets inside lists.
+		{
+			ValueModifiers: []ValueModifier{
+				testModifier1{},
+			},
+			IgnoreFields: []string{},
+			Input: `{
+  "list1": [
+    {
+      "pass1": "pass1"
+    }
+  ]
+}`,
+			Expected: `{
+  "list1": [
+    {
+      "pass1": "pass1-modified1"
+    }
+  ]
+}`,
+		},
+		// Test case 4, a single modifier modifies all secrets, but ignores the ones
+		// configured using IgnoreFields.
+		{
+			ValueModifiers: []ValueModifier{
+				testModifier1{},
+			},
+			IgnoreFields: []string{
+				"noSecret1",
+			},
+			Input: `{
+  "noSecret1": "foo",
+  "pass1": "pass1"
+}`,
+			Expected: `{
+  "noSecret1": "foo",
+  "pass1": "pass1-modified1"
+}`,
+		},
+		// Test case 5, multiple modifiers modify all secrets, but ignore the ones
+		// configured using IgnoreFields.
+		{
+			ValueModifiers: []ValueModifier{
+				testModifier1{},
+				testModifier2{},
+			},
+			IgnoreFields: []string{
+				"noSecret1",
+				"noSecret2",
+			},
+			Input: `{
+  "noSecret1": "foo",
+  "noSecret2": "bar",
+  "pass1": "pass1",
+  "pass2": "pass2"
+}`,
+			Expected: `{
+  "noSecret1": "foo",
+  "noSecret2": "bar",
+  "pass1": "pass1-modified1-modified2",
+  "pass2": "pass2-modified1-modified2"
+}`,
+		},
+		// Test case 6, nested blocks, multiple modifiers modify all secrets, but
+		// ignore the ones configured using IgnoreFields.
+		{
+			ValueModifiers: []ValueModifier{
+				testModifier1{},
+				testModifier2{},
+			},
+			IgnoreFields: []string{
+				"noSecret1",
+				"noSecret2",
+			},
+			Input: `{
+  "block1": {
+    "block11": {
+      "pass1": "pass1"
+    },
+    "pass2": "pass2"
+  },
+  "block2": {
+    "block21": {
+      "pass3": "pass3"
+    },
+    "pass4": "pass4"
+  },
+  "noSecret1": "foo",
+  "noSecret2": "bar",
+  "pass5": "pass5",
+  "pass6": 123456
+}`,
+			Expected: `{
+  "block1": {
+    "block11": {
+      "pass1": "pass1-modified1-modified2"
+    },
+    "pass2": "pass2-modified1-modified2"
+  },
+  "block2": {
+    "block21": {
+      "pass3": "pass3-modified1-modified2"
+    },
+    "pass4": "pass4-modified1-modified2"
+  },
+  "noSecret1": "foo",
+  "noSecret2": "bar",
+  "pass5": "pass5-modified1-modified2",
+  "pass6": "123456-modified1-modified2"
+}`,
```

Comment:
I would rather like to prevent testing all invalid cases now. Do you have a list of tests that you care about? Something more concrete? I feel like I already did some decent testing. I would like to extend this based on real world problems. Lets keep it this way that we iterate on our code instead of making it perfect in the first place. This will never work out anyway.